### PR TITLE
[Enhancement] Improve storage volume hdfs backward compatibility

### PR DESCRIPTION
**Pull request by TszKitLo40**
Fixes #issue

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/credential/hdfs/HDFSCloudCredential.java [hunk 1]
```diff
@@ -99,6 +105,10 @@ public String getCredentialString() {
     @Override
     public FileStoreInfo toFileStoreInfo() {
         // TODO: support hdfs credential
-        return null;
+        FileStoreInfo.Builder fileStore = FileStoreInfo.newBuilder();
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/credential/hdfs/HDFSCloudCredential.java:108]: remove the `TODO:` comment.

File: fe/fe-core/src/test/java/com/starrocks/storagevolume/StorageVolumeTest.java [hunk 2]
```diff
@@ -231,6 +231,15 @@ public void testHDFSKerberosCredential() throws AnalysisException {
         Assert.assertEquals(5, hdfsCloudConfiguration.getHdfsCloudCredential().getHaConfigurations().size());
     }
 
+    @Test
+    public void testHDFSEmptyCredential() {
+        Map<String, String> storageParams = new HashMap<>();
+        StorageVolume sv = new StorageVolume("1", "test", "hdfs", Arrays.asList("hdfs://abc"),
+                storageParams, true, "");
+        CloudConfiguration cloudConfiguration = sv.getCloudConfiguration();
+        Assert.assertEquals(CloudType.HDFS, cloudConfiguration.getCloudType());
```
Comment (xiangguangyxg) [inline fe/fe-core/src/test/java/com/starrocks/storagevolume/StorageVolumeTest.java:240]: check authentication is empty ?

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 7 / 13 (53.85%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/credential/hdfs/HDFSCloudConfiguration.java|0|1|00.00%|[58]|
|:large_blue_circle:|com/starrocks/credential/hdfs/HDFSCloudCredential.java|2|7|28.57%|[109, 110, 111, 112, 113]|
|:large_blue_circle:|com/starrocks/server/SharedDataStorageVolumeMgr.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/storagevolume/StorageVolume.java|4|4|100.00%|[]|

**Review by Smith-Cruise — APPROVED** [review]
Help to reorder aliyun and hdfs create order in CloudConfigurationFactory is better.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 15 / 15 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/credential/hdfs/HDFSCloudConfiguration.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/SharedDataStorageVolumeMgr.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/credential/CloudConfigurationFactory.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/credential/hdfs/HDFSCloudCredential.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/storagevolume/StorageVolume.java|4|4|100.00%|[]|

**Review by xiangguangyxg — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Review by sduzh — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Pull request merged**